### PR TITLE
docs: add metadata for parser/processor

### PR DESCRIPTION
**Pull request by JLHwung**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Added `metadata` descriptions to `custom-parsers.md` and `custom-processors.md`. I don't know if `metadata` object is also implemented for formatters, if so I am happy to add a section covering that, too.

#### Is there anything you'd like reviewers to focus on?

The current `meta` sections are mostly copied from `plugins.md`. Frankly I don't know if `meta` object helps ESLint cache the parsers/processors. If not please correct me.

### CODE REVIEW

File: docs/src/extend/custom-parsers.md [hunk 1]
```diff
@@ -12,6 +12,24 @@ ESLint custom parsers let you extend ESLint to support linting new non-standard
 
 ## Creating a Custom Parser
 
+### Metadata in Custom Parsers
```
Comment (nzakas) [inline docs/src/extend/custom-parsers.md:15]:
```suggestion
### Meta Data in Custom Parsers
```

File: docs/src/extend/custom-parsers.md [hunk 2]
```diff
@@ -12,6 +12,24 @@ ESLint custom parsers let you extend ESLint to support linting new non-standard
...
+        version: "1.2.3"
+    }
+};
+```
+
+The `meta.name` property should match the npm package name for your custom parser and the `meta.version` property should match the npm package version for your custom parser. The easiest way to accomplish this is by reading this information from your `package.json`.
+
+### Methods in Custom Parsers
+
 A custom parser is a JavaScript object with either a `parse` or `parseForESLint` method. The `parse` method only returns the AST, whereas `parseForESLint` also returns additional values that let the parser customize the behavior of ESLint even more.
```
Comment (nzakas) [inline docs/src/extend/custom-parsers.md:33]: Can you rearrange this so the meta data section comes after these other sections? We'd like people to focus on the methods first and then think about meta data.

**Review by nzakas — CHANGES_REQUESTED** [review]

File: docs/src/extend/custom-parsers.md (hunk 2 above)
Comment (JLHwung) [reply docs/src/extend/custom-parsers.md:17]: Np. I copied the ordering from `plugins.md`. Also updated the ordering there.

**Review by harish-sethuraman — APPROVED** [review]
LGTM. Thanks

**Pull request merged**